tests/tests.js

```diff
@@ -519,9 +519,228 @@ describe('i18n tests', function(){
+                region: "US",
+                quality: 1.0
+            }], {
+                    required: [
+                        'Name',
+                        'Description'
+                    ],
+                    optional: [
+                        'PriceBand'
+                    ]
+                });
+
+            (result.translations.PriceBand == null).should.be.true;
+            (result.localization.PriceBand == null).should.be.true;
+        });
+
+        it('should fallback to a specfied secondary locale', function () {
+            var result = i18n.transformByField(translations, [{
+                code: "it",
+                region: "IT",
+                quality: 1.0
+            },
+                {
+                    code: "en",
+                    region: "US",
+                    quality: 0.9
+                }], {
+                    required: [
+                        'Name',
+                        'Description'
+                    ],
+                    optional: []
+                });
+
+            result.translations.Name.should.eql('Benvenuti in un mondo di cibo');
+            result.translations.Description.should.eql('This is a description in US English');
+
+            result.localization.Name.should.eql('it-IT');
+            result.localization.Description.should.eql('en-US');
+        });
+
+        it('should fallback to primary locale for required field if locale not present', function () {
+            var result = i18n.transformByField(translations, [{
+                code: "es",
+                region: "ES",
+                quality: 1.0
+            }], {
+                    required: [
+                        'Name',
+                        'Description'
+                    ],
+                    optional: [
+                        'PriceBand'
+                    ]
+                });
+
+            result.translations.Name.should.eql('gonna drink some beer and shoot some stuff y\'all');
+            result.translations.Description.should.eql('This is a description in US English');
+
+            result.localization.Name.should.eql('en-US');
+            result.localization.Description.should.eql('en-US');
+
+            (result.translations.PriceBand == null).should.be.true;
+            (result.localization.PriceBand == null).should.be.true;
+        });
+
+        it('should not fallback to primary locale for optional field if not present', function () {
+            var result = i18n.transformByField(translations, [{
+                code: "it",
+                region: "IT",
+                quality: 1.0
+            }], {
+                    required: [],
+                    optional: [
+                        'PriceBand'
+                    ]
+                });
+
+            (result.translations.PriceBand === null).should.be.true;
+            (result.localization.PriceBand === null).should.be.true;
+        });
+    });
+});
```

Comment:
Can we add one case (or more) where one of the accept-languages does not specify a region (e.g. 'en') with some of the fields being in 'en-US' for example)?